src/lib/controllers/DevicesController.js

```diff
@@ -98,6 +130,7 @@ console.log(postBody);
         postBody,
       );
 
+      // TODO add userID checking
```

Comment:
You should do this next.  That way the logic is in one place.. In DeviceRepository you do some logic there which I think could be moved to DeviceAttributeRepository.